lib/galaxy/datatypes/infernal.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import os
+
+from galaxy.datatypes.data import Text
+from galaxy.datatypes.data import get_file_peek
+
+from galaxy.datatypes.metadata import MetadataElement
+
+log = logging.getLogger(__name__)
+
+
+class Infernal_CM_1_1(Text):
+    file_ext = "cm"
+
+    MetadataElement(name="number_of_models", default=0, desc="Number of covariance models", readonly=True, visible=True, optional=True, no_value=0)
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = get_file_peek(dataset.file_name, is_multi_byte=is_multi_byte)
+            if (dataset.metadata.number_of_models == 1):
+                dataset.blurb = "1 model"
+            else:
+                dataset.blurb = "%s models" % dataset.metadata.number_of_models
+            dataset.peek = get_file_peek(dataset.file_name, is_multi_byte=is_multi_byte)
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disc'
+
+    def sniff(self, filename):
+        with open('myfile.txt', 'r') as f:
+            first_line = f.readline()

```

Comment:
you could use here read(11) to read the first characters and compare it with your string

Reply:
'myfile.txt' must be wrong or filename is unnecessary. agree?

Reply:
Yes this needs to be filename I guess.